api/src/test/java/org/openmrs/api/UserServiceTest.java

```diff
@@ -1196,4 +1198,108 @@ public void saveUserProperties_shouldRemoveAllExistingPropertiesAndAssignNewProp
+		String oldPassword = null;
+		String newPassword = "newPasswordString123";
+		userService.changePassword(user6001, oldPassword, newPassword);
+		//remember previously logged user
+		User previouslyLogged = Context.getAuthenticatedUser();
+		//authenticated user has change user passwords priviliges
+		Context.authenticate(user6001.getUsername(), newPassword);
+		//clean up
+		Context.logout();
+		Context.authenticate("admin", "test");
+	}
+	
+	/**
+	 * @see UserService#changePassword(User,String,String)
+	 * @verifies throw APIException if old password is not correct
+	 */
+	@Test(expected = APIAuthenticationException.class)
+	public void changePassword_shouldThrowAPIExceptionIfOldPasswordIsNotCorrect() throws Exception {
+		executeDataSet(XML_FILENAME_WITH_DATA_FOR_CHANGE_PASSWORD_ACTION);
+		final UserService userService = Context.getUserService();
+		//user 6001 has password userServiceTest
+		User user6001 = userService.getUser(6001);
+		String wrongPassword = "wrong password!";
+		String newPassword = "newPasswordString";
+		//log in user without change user passwords privileges
+		//user6001 has not got required priviliges
+		Context.authenticate(user6001.getUsername(), "userServiceTest");
+		userService.changePassword(user6001, wrongPassword, newPassword);
+	}
+	
+	/**
+	 * @see UserService#changePassword(User,String,String)
+	 * @verifies throw exception if oldPassword is null and changing user have not privileges
+	 */
+	@Test(expected = APIException.class)
+	public void changePassword_shouldThrowExceptionIfOldPasswordIsNullAndChangingUserHaveNotPrivileges() throws Exception {
+		executeDataSet(XML_FILENAME_WITH_DATA_FOR_CHANGE_PASSWORD_ACTION);
+		final UserService userService = Context.getUserService();
+		//user 6001 has password userServiceTest
+		User user6001 = userService.getUser(6001);
+		String oldPassword = null;
+		String newPassword = "newPasswordString";
+		//log in user without change user passwords privileges
+		//user6001 has not got required priviliges
+		Context.authenticate(user6001.getUsername(), "userServiceTest");
+		userService.changePassword(user6001, oldPassword, newPassword);
+	}
+	
+	/**
+	 * @see UserService#changePassword(User,String,String)
+	 * @verifies throw exception if new password is too shot
+	 */
+	@Test(expected = ShortPasswordException.class)
+	public void changePassword_shouldThrowExceptionIfNewPasswortIsTooShort() throws Exception {
+		executeDataSet(XML_FILENAME_WITH_DATA_FOR_CHANGE_PASSWORD_ACTION);
+		final UserService userService = Context.getUserService();
+		//user 6001 has password userServiceTest
+		User user6001 = userService.getUser(6001);
+		String oldPassword = "userServiceTest";
+		String weakPassword = "weak";
+		userService.changePassword(user6001, oldPassword, weakPassword);
+	}
+	
+	/**
+	 * @see UserService#changePassword(User,String,String)
+	 * @verifies throw APIException if given user does not exist
+	 */
+	@Test(expected = APIException.class)

```

Comment:
We prefer to use Junit's ExpectedException approach since it also lets you check the actual error message because the exception could be from somewhere else especially for APIException which is throw from very many places. Can you do the same for the other tests where you expect and an APIException